File: java/Mahasiswa.java

/*
Saya [Elsa Nabiilah] [2108805] mengerjakan LATIHAN 2 dalam mata kuliah Desain Pemrograman Berorientasi Objek
untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan.
Aamiin.
*/

//membuat class
public class Mahasiswa extends Human {
    private String nim;
    private String nama;
    private String jk;
    private String fak;
    private String prodi;

    // initialize
    public Mahasiswa() {
        this.nim = "";
        this.nama = "";
        this.jk = "";
        this.fak = "";
        this.prodi = "";
    }

    public Mahasiswa(String nim, String nama, String jk, String fak, String prodi) {
        this.nim = nim;
        this.nama = nama;
        this.jk = jk;
        this.fak = fak;
        this.prodi = prodi;
    }

    // getter setter method
    public String getnim() {
        return this.nim;
    }

    public void setnim(String nim) {
        this.nim = nim;
    }

    public String getnama() {
        return this.nama;
    }

    public void setnama(String nama) {
        this.nama = nama;
    }

    public String getjk() {
        return this.jk;
    }

    public void setjk(String jk) {
        this.jk = jk;
    }

    public String getfak() {
        return this.fak;
    }

    public void setfak(String fak) {
        this.fak = fak;
    }

    public String getprof() {
        return this.prodi;
    }

    public void setprodi(String prodi) {
        this.prodi = prodi;
    }

    // function
    void nim() {
        System.out.println("NIM           : " + this.nim);

    }

    void fak() {
        System.out.println("Fakultas      : " + this.fak);
    }

    void prod() {
        System.out.println("Program Studi : " + this.prodi);
    }

    void nm() {
        System.out.println("Nama          : " + this.nama);
    }

    void jenisk() {
        System.out.println("Jenis Kelamin : " + this.jk);
    }

}

File: cpp/Mahasiswa.cpp
#include <bits/stdc++.h>
using namespace std;
#include "Human.cpp"
/*
Saya [Elsa Nabiilah] [2108805]
mengerjakan LATIHAN 2 dalam mata kuliah Desain Pemrograman Berorientasi Objek
untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan.
Aamiin.
*/
class Mahasiswa : public Human
{
    // class mahasiswa mejadi child/toodler dari class human karena
    // atribut Mahasiswa melekat kaitannya dengan atribut Human karena sama - sama mengarah
    // kepada point-point identitas
private: // deklarasi private atribut
    string nim;

    string fak;

    string prodi;

public:
    // contructor
    Mahasiswa()
    {
    }
    Mahasiswa(string nim, string fak, string prodi)
    {
        this->nim = nim;
        this->fak = fak;
        this->prodi = prodi;
    }
    // setter getter method
    string getnim()
    {
        // getter nim
        return this->nim;
    }

    void setnim(string nim)
    {
        // setter nim
        this->nim = nim;
    }

    string getfak()
    {
        // getter fakultas
        return this->fak;
    }

    void setfak(string fak)
    { // setter fakultas
        this->fak = fak;
    }

    string getprof()
    {
        // getter prodi
        return this->prodi;
    }

    void setprodi(string prodi)
    { // setter prodi
        this->prodi = prodi;
    }

    // function untuk menampilkan per atribut
    void printnim()
    {
        cout << "NIM           : " << this->getnim() << endl;
    }
    void printps()
    {
        cout << "Program Studi : " << this->getprof() << endl;
    }
    void printfak()
    {
        cout << "Fakultas      : " << this->getfak() << endl;
    }

    ~Mahasiswa() {}
};
